mautic/mautic
# Update Sendgrid library to v7 and add hash_id parameter for callback TPROD-101

…tracking

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | 3.2
| Bug fix?                               | yes
| New feature?                           | yes
| Deprecations?                          | no
| Issue(s) addressed                     | Fixes #5967  




#### Description:
Sendgrid library was updated to v7 with some differences in how CC and BCC are included in the payload. Plus, it can include other parameters that we use to track the failed statistics from SendGrid.


#### Steps to test this PR - BCC:
1. Load up [this PR](https://mautibox.com)
2. Composer install
3. Send an email with BCC
4. Check that email in BCC was received

#### Steps to test this PR - Failed tracking:
1. Load up [this PR](https://mautibox.com)
2. Composer install
3. Configure callback URL in SendGrid (not localhost)
4. Send an email with a wrong email address (for failed tracking check)
5. Wait for the wrong email address contact to update
